pkg/common/jobcontroller/pod.go

```diff
@@ -113,12 +113,6 @@ func (jc *JobController) UpdatePod(old, cur interface{}) {
 // obj could be an *v1.Pod, or a DeletionFinalStateUnknown marker item.
 func (jc *JobController) DeletePod(obj interface{}) {
 	pod, ok := obj.(*v1.Pod)
-	if pod == nil {
```

Comment:
Why do we remove the check? Will https://github.com/kubeflow/tf-operator/pull/1015/files#diff-0649a2a405335166e6ddd97221ff274aR140 report nil pointer exception?

Reply:
I think the panic occur in this line 
```
logger := jclogger.LoggerForPod(pod, jc.Controller.GetAPIGroupVersionKind().Kind)
```
if pod is nil, that is to say the type assertion is not ok, we should go to the logic that process `DeletedFinalStateUnknown`, if it is not ok again, we just return. So just move the logger line to 134.

Reply:
Gotcha, reasonable, LGTM

/lgtm